src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/ComponentAccessibilityCodeActionProvider.cs

```diff
@@ -39,21 +38,28 @@ public ComponentAccessibilityCodeActionProvider(TagHelperFactsService tagHelperF
         using var _ = ListPool<RazorVSInternalCodeAction>.GetPooledObject(out var codeActions);
 
         // Locate cursor
-        var change = new SourceChange(context.Location.AbsoluteIndex, length: 0, newText: string.Empty);
-        var node = context.CodeDocument.GetSyntaxTree().Root.LocateOwner(change);
+        var node = context.CodeDocument.GetSyntaxTree().Root.FindInnermostNode(context.Location.AbsoluteIndex);
         if (node is null)
         {
             return s_emptyResult;
         }
 
         // Find start tag. We allow this code action to work from anywhere in the start tag, which includes
         // embedded C#, so we just have to traverse up the tree to find a start tag if there is one.
-        var startTag = (MarkupStartTagSyntax?)node.Ancestors().FirstOrDefault(n => n is MarkupStartTagSyntax);
+        var startTag = (MarkupStartTagSyntax?)node.AncestorsAndSelf().FirstOrDefault(n => n is MarkupStartTagSyntax);
```

Comment:
There are a few places around the IDE that are unnecessarily allocation enumerables by using this pattern. I'll send a quick PR to address the other cases.

```suggestion
        var startTag = (MarkupStartTagSyntax?)node.FirstAncestorOrSelf<SyntaxNode>(n => n is MarkupStartTagSyntax);
```

Reply:
https://github.com/dotnet/razor/pull/9053 for that other refactoring.